fl4health/strategies/flash.py

```diff
@@ -100,26 +105,50 @@ def __init__(
             initial_parameters=initial_parameters,
             fit_metrics_aggregation_fn=fit_metrics_aggregation_fn,
             evaluate_metrics_aggregation_fn=evaluate_metrics_aggregation_fn,
-            eta=eta,
-            eta_l=eta_l,
-            beta_1=beta_1,
-            beta_2=beta_2,
-            tau=tau,
+            weighted_aggregation=weighted_aggregation,
+            weighted_eval_losses=weighted_eval_losses,
         )
+        self.current_weights = parameters_to_ndarrays(initial_parameters)
+        self.eta = eta
+        self.eta_l = eta_l
+        self.tau = tau
+        self.beta_1 = beta_1
+        self.beta_2 = beta_2
+        self.m_t: Optional[NDArrays] = None
+        self.v_t: Optional[NDArrays] = None
         self.d_t: Optional[NDArrays] = None
 
     def __repr__(self) -> str:
         """Compute a string representation of the strategy."""
-        rep = f"FLASH(accept_failures={self.accept_failures})"
+        rep = f"Flash(accept_failures={self.accept_failures})"
         return rep
 
+    def _update_parameters(self, delta_t: NDArrays) -> None:
+        """Update m_t, v_t, beta_3, and d_t per-element"""
+        assert self.v_t is not None and self.m_t is not None and self.d_t is not None
+
+        for i, (delta, m, v, d_prev) in enumerate(zip(delta_t, self.m_t, self.v_t, self.d_t)):
```

Comment:
Super minor, but I'd attach the `_prev` prefix to `m` and `v` here too just to be clear that they are the "previous values"